src/ucp/tag/rndv.c

```diff
@@ -839,8 +839,12 @@ UCS_PROFILE_FUNC_VOID(ucp_rndv_matched, (worker, rreq, rndv_rts_hdr),
         goto out;
     }
 
-    rndv_req->send.ep           = ucp_worker_get_ep_by_ptr(worker,
-                                                           rndv_rts_hdr->sreq.ep_ptr);
+    rndv_req->send.ep = ucp_worker_get_ep_by_ptr(worker, rndv_rts_hdr->sreq.ep_ptr);
+    if (rndv_req->send.ep == NULL) {
+        ucp_request_put(rndv_req);
```

Comment:
1. need to complete receive request with status cancel
2. can check this before allocating rndv_req